### PR TITLE
fix: canvas resizing

### PR DESCRIPTION
**Pull request by mauroporras**

### CODE REVIEW

**Comment by mauroporras** [comment]
@phtaylor, it contains breaking changes, I wanted to run it by you first.

**Comment by phtaylor** [comment]
Does this make it a major version change then?

**Comment by mauroporras** [comment]
> Does this make it a major version change then?

Only if we decide so. Things like using a DIV as root or a static canvas, are going to break, because we're now observing the parent's resizing event.

**Comment by mauroporras** [comment]
@phtaylor, pushed without breaking change.

**Comment by phtaylor** [comment]
@mauroporras , Maybe renderer classses do provide a teardown function called 'destroy'

here is an example. 
https://github.com/ZeaInc/zea-engine/blob/master/src/Renderer/GLTexture2D.js#L699

This would need to be explicitly called to clean up the renderer. The Renderer should then call destroy on any resources it owns that need to be cleaned up. For now, lets just add a 'destroy' method to the GLBaseRenderer for the resize observer.

File: .prettierrc.yaml [hunk 1]
```diff
@@ -1,3 +1,2 @@
 semi: false
 singleQuote: true
-printWidth: 120
```
Comment (mauroporras) [inline .prettierrc.yaml:3]: Please ignore. Will be reverted.

File: package.json [hunk 2]
```diff
@@ -31,7 +31,7 @@
     "test:coverage": "jest --coverage",
     "test:debug": "node --inspect ./node_modules/jest/bin/jest.js --runInBand --watch",
     "test:watch": "jest --watch",
-    "test:e2e": "percy exec cypress run --browser chrome --headless",
+    "test:e2e": "percy exec cypress run --browser chrome --headless --spec cypress/integration/transparent-geoms-sorting-tinyscene-spec.js",
```
Comment (mauroporras) [inline package.json:34]: Please ignore. Will be reverted.

**Pull request merged**